nibabel/nifti1.py

```diff
@@ -2443,9 +2443,13 @@ def _get_analyze_compat_dtype(arr):
         return np.dtype('int16' if arr.max() <= np.iinfo(np.int16).max else 'int32')
 
     mn, mx = arr.min(), arr.max()
-    if np.can_cast(mn, np.int32) and np.can_cast(mx, np.int32):
+    if (isinstance(mn, int) and isinstance(mx, int)) or (
```

Comment:
The problem with this is that on most platforms Python `int` is 64-bit. So you could have a value of `2 ** 33` represented properly using it. But saving in `np.int32` is not safe. So I think this check needs to change to something involving `np.iinfo('int32')` max/min :disappointed: